APEX-184

vrozov opened the pull request:
@tweise, @PramodSSImmaneni, please pull into release-3.2 once #121 is merged.

243826 commented on dt_checks.xml:
```diff
@@ -10,4 +10,110 @@
+    <module name="RightCurly">
+      <property name="option" value="alone"/>
+      <property name="tokens"
+                value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
```
is there a difference when you put a space after `,` and no space after `,`?

chandnisingh replied:
No will fix that

243826 commented on dt_checks.xml:
```diff
@@ -10,4 +10,110 @@
+      <message key="ws.preceded"
+               value="GenericWhitespace ''{0}'' is preceded with whitespace."/>
+      <message key="ws.illegalFollow"
+               value="GenericWhitespace ''{0}'' should followed by whitespace."/>
```
`be ` missing?

chandnisingh replied:
Will fix it

243826 commented:
overall it looks good to me. Can you enable it such that only if more deviations are added, the build will fail? And make it chatty only when a system property is specified while building.

chandnisingh commented:
Will count existing deviations and will add a check that if more are introduced then build will fail. 
Also should we add a check for license?

243826 commented:
check for license is already there (mvn license:check i think). And I think
that will be more comprehensive. But if Checkstyle does at least as good of
a job, I think it may warrant consolidation.

On Mon, Aug 24, 2015 at 1:00 PM, Chandni Singh <notifications@github.com>
wrote:

> Will count existing deviations and will add a check that if more are
> introduced then build will fail.
> Also should we add a check for license?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DataTorrent/Apex/pull/124#issuecomment-134359318>.
>

Commit:
APEX-184 #resolve
1. In autoFlushExecutor don't exit run() until there is at least one listener that has more data to send.
2. Do not enable read in resumeReadIfSuspended when not able to switch to a new buffer.
3. Fix possible race condition in Block acquire.
4. Fix for incorrect counting of in memory block permits.
5. Fix check style violations.
